massa-client/src/tests/scenarios.rs

```diff
@@ -15,25 +21,40 @@ async fn run_client_cmd(cmd: &str) -> Result<String> {
                 .stdout,
         )?
         .to_string();
+        println!("{}", output);
         // TODO: compare with tcp connect error as a JSON object?
         if !output.contains("tcp connect error") {
             return Ok(output);
         }
         tokio::time::sleep(std::time::Duration::from_secs(TIMEOUT)).await;
     }
-    bail!("tcp connect error")
+    bail!("was not able to send command")
+}
+
+fn run_node(duration: Duration) -> JoinHandle<String> {
+    // TODO: do we have a better way to spawn `massa_node`?
+    std::thread::spawn(|| {
+        std::str::from_utf8(
+            &Command::new("cargo")
+                .args(["run", "--feature", "test"]) // TODO add at least genesis timestamp to a test feature
+                .current_dir("../massa-node")
+                .timeout(duration)
+                .assert()
+                .get_output()
+                .stdout,
+        )
+        .unwrap()
+        .to_string()
+    })
 }
 
 #[tokio::test]
 #[serial]
-async fn run_node() {
-    // TODO: @adrien-zinger do we have a better way to spawn `massa_node`?
-    std::thread::spawn(|| {
-        Command::new("cargo")
-            .arg("run")
-            .current_dir("../massa-node")
-            .unwrap()
-    });
+async fn test_run_node() {
+    let handle = run_node(Duration::from_secs(60 * 3));
+    sleep(Duration::from_secs(180)); // let it compile and start
+    let a = handle.join().unwrap();
```

Comment:
But but ... I didn't get it ... if you join `massa-node` on the first `#[tokio test]` `#[serial]`, how do you run the full test suite on the same running node?

Reply:
I just want that test to run once and then I will fill up that with the scenario

Reply:
You want to write the whole client scenario inside the same test? Okay, I wasn’t thinking about that, and yes you're right that’s better :)

Reply:
For some commands we re use part of the output of the previous ones